src/app/components/category-products/category-products.component.html

```diff
@@ -27,7 +27,8 @@ <h5 class="mb-3">Price</h5>
         <div class="mb-8">
           <h5 class="mb-3">Rating</h5>
           <div *ngFor="let item of ratingOptions$ | async" class="form-check mb-2">
-            <input [id]="item" class="form-check-input" type="radio" [name]="item" (change)="onRatingChange(item)" [checked]="(rating$ | async) === item">
+            <input [id]="item" class="form-check-input" type="radio" [name]="item" (change)="onRatingChange(item)"
+              [checked]="(rating$ | async) === item">
```

Comment:
In my opinion, you should move this "rating$ | async" higher and use it here. In your case you subscribe to this for each ratingOptions$ element and the value doesn't change.